governator-test-junit/src/main/java/com/netflix/governator/guice/test/AnnotationBasedTestInjectorManager.java

```diff
@@ -0,0 +1,223 @@
+        for (Element element : elements) {
+            element.acceptVisitor(new DefaultElementVisitor<Void>() {
+                @Override
+                public <T> Void visit(Binding<T> binding) {
+                    final Binding<T> finalBinding = binding;
+                    if (spyTargets.contains(binding.getKey())) {
+                        AbstractModule spyModule = new AbstractModule() {
+                            protected void configure() {
+                                final String finalBindingName = "Spied "
+                                        + (finalBinding.getKey().getAnnotation() != null
+                                                ? finalBinding.getKey().getAnnotation().toString() : "")
+                                        + finalBinding.getKey().getTypeLiteral();
+                                final Key<T> newUniqueKey = Key.get(finalBinding.getKey().getTypeLiteral(),
+                                        Names.named(finalBindingName));
+                                finalBinding.acceptTargetVisitor(new CopyBindingTargetVisitor<>(binder().bind(newUniqueKey)));
+                                bind(finalBinding.getKey()).toProvider(new SingletonProvider<T>() {
+                                    @Inject
+                                    Injector injector;
+
+                                    protected T create() {
+                                        T t = (T) injector.getInstance(newUniqueKey);
+                                        return Mockito.spy(t);
+                                    };
+                                });
+                            }
+                        };
+                        overrideModules.add(spyModule);
+                    }
+                    return null;
+                }
+            });
+        }
+    }
+    
+    public void prepareConfigForTestClass(Class<?> testClass, Method testMethod) {
+        for(Class<?> parentClass : getAllSuperClassesInReverseOrder(testClass)) {
+            classLevelOverrides.setProperties(testPropertyOverrideAnnotationReader.getPropertiesForAnnotation(parentClass.getAnnotation(TestPropertyOverride.class)));
+        }
+        classLevelOverrides.setProperties(testPropertyOverrideAnnotationReader.getPropertiesForAnnotation(testClass.getAnnotation(TestPropertyOverride.class)));
+        methodLevelOverrides.setProperties(testPropertyOverrideAnnotationReader.getPropertiesForAnnotation(testMethod.getAnnotation(TestPropertyOverride.class)));                    
+    }
+    
+    public void cleanUpMethodLevelConfig() {
+        testCompositeConfig.resetForTest();
+    }
+    private List<Class<?>> getAllSuperClassesInReverseOrder(Class<?> clazz) {
+        List<Class<?>> allSuperclasses = ClassUtils.getAllSuperclasses(clazz);
+        Collections.reverse(allSuperclasses);
+        return allSuperclasses;
+    }
+
+    private class MockitoOverrideModule<T> extends AbstractModule {
```

Comment:
Not static as there is state being tracked within this module. Not a great thing, but I needed to keep hold of the list of things being mocked for cleanup purposes.